docs/writing-your-first-rclrs-node.md

```diff
@@ -0,0 +1,214 @@
+
+So, to store the received data in the struct, the following things have to change:
+1. Import `Mutex`
+2. Adjust the type of the `data` field
+3. Create two pointers to the same data (wrapped in a `Mutex`)
+4. Make the closure `move`, and inside it, lock the `Mutex` and store the message  
+
+```rust
+use std::sync::{Arc, Mutex};  // (1)
+use std_msgs::msg::String as StringMsg;
+
+struct RepublisherNode {
+    node: rclrs::Node,
+    _subscription: Arc<rclrs::Subscription<StringMsg>>,
+    data: Arc<Mutex<Option<StringMsg>>>,  // (2)
+}
+
+impl RepublisherNode {
+    fn new(context: &rclrs::Context) -> Result<Self, rclrs::RclrsError> {
+        let mut node = rclrs::Node::new("republisher", context)?;
+        let data = Arc::new(Mutex::new(None));  // (3)
+        let data_cb = Arc::clone(&data);
+        let _subscription = {
+            // Create a new shared pointer instance that will be owned by the closure
+            node.create_subscription(
+                "in_topic",
+                rclrs::QOS_PROFILE_DEFAULT,
+                move |msg: StringMsg| {
+                    // This subscription now owns the data_cb variable
+                    *data_cb.lock().unwrap() = Some(msg);  // (4)
+                },
+            )?
+        };
+        Ok(Self {
+            node,
+            _subscription,
+            data,
+        })
+    }
+}
+```
+
+If that seems needlessly complicated – maybe it is, in the sense that `rclrs` could potentially introduce new abstractions to improve the ergonomics of this use case. This is to be discussed.
+
+If you couldn't follow the explanation involving borrowing, closures etc. above, an explanation of these concepts is unfortunately out of scope of this tutorial. There are many good Rust books and tutorials that can help you understand these crucial features.
+
+## Periodically run a republishing function
+
+The node still doesn't republish the received messages. First, let's add a publisher to the node:
+
+```rust
+// Add this new field to the RepublisherNode struct:
+publisher: rclrs::Publisher<StringMsg>,
+
+// Change the end of RepublisherNode::new() to this:
+let publisher = node.create_publisher("out_topic", rclrs::QOS_PROFILE_DEFAULT)?;
+Ok(Self {
+    node,
+    _subscription,
+    publisher,
+    data,
+})
+```
+
+Then, let's add a `republish()` function to the `RepublisherNode` that publishes the latest message received, or does nothing if none was received:
+
+```rust
+fn republish(&self) -> Result<(), rclrs::RclrsError> {
+    if let Some(s) = &*self.data.lock().unwrap() {
+        self.publisher.publish(s)?;
+    }
+    Ok(())
+}
+```
+
+What's left to do is to call this function every second. `rclrs` doesn't have built-in functionality to do run a function at a fixed interval, but it's easy enough to achieve with a thread and the sleep function. Add `use std::time::Duration;` to the imports, and change your main function to spawn a separate thread:
```

Comment:
All other code changes in this tutorial are called out with small blocks (instead of being inline) even when they are one-line changes. For consistency maybe consider changing this addition to also be a code block?

Reply:
I agree. I moved it into the main function, so that the changes aren't so scattered. I like these "local" `use` statements better anyway when there's only one place that it will be used in.